projects/igniteui-angular/src/lib/grids/headers/sorting-index.pipe.ts

```diff
@@ -0,0 +1,20 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { ISortingExpression } from '../../data-operations/sorting-expression.interface';
+import { IgxGridHeaderComponent } from './grid-header.component';
+
+@Pipe({
+  name: 'sortingIndex',
+  pure: true
+})
+export class SortingIndexPipe implements PipeTransform {
+
+  constructor(private igxHeader: IgxGridHeaderComponent) {}
+  transform(columnField: string, sortingExpressions: ISortingExpression[]): number {
+    const grid = this.igxHeader.grid;
```

Comment:
If you need the grid instance in the pipe just pass it as a parameter in the pipe `transform` method.
Don't see the point of injecting the header component just to get a grid instance.